w-12273311-Reuse types first draft (#235)
mulesoft/docs-dataweave: modules/ROOT/pages/dataweave-type-reuse-modules.adoc

= Reusing Types from DataWeave Modules
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> Reuse types declared in other DataWeave modules by using the import directive:

Reviewer: ```suggestion
To reuse types declared in other DataWeave modules, use the import directive:
```

Reviewer: Does import need a font change?

Author: Reworded + added backticks to `import`.

> .DataWeave Module (dw/Weather.dwl)

Reviewer 2: ```suggestion
.DataWeave Module (`dw/Weather.dwl`):
```

[source,dataweave,linenums]
----
%dw 2.0
type Weather = "cloudy" | "sunny" | "rainy" | "stormy"
type DetailedWeather = {temperature: Number, weather: Weather}
----

> You can use those types as any other type in DataWeave, as constraints for variables and functions or for pattern matching and checking types at runtime:

Reviewer: "those" -> imported?

I kind of want something before "as". "the same as"?

Author: Added imported.
